File: 0208-implement-trie-prefix-tree/0208-implement-trie-prefix-tree.cpp
class Node{
public:
    Node *chars[26];
    bool end=false;
};
class Trie {
public:
    Node *root = new Node();
    Trie() {
        
    }
    
    void insert(string word) {
        Node *node = root;
        
        for(char c:word){
            if(!node->chars[c-'a'])
                node->chars[c-'a'] = new Node();
            node = node->chars[c-'a'];
        }
        node->end=true;
    }
    
    bool search(string word) {
        Node *node = root;
        
        for(char c:word){
            if(!node->chars[c-'a'])
                return false;
            node = node->chars[c-'a'];
        }
        return node->end;
    }
    
    bool startsWith(string prefix) {
        Node *node = root;
        
        for(char c:prefix){
            if(!node->chars[c-'a'])
                return false;
            node = node->chars[c-'a'];
        }
        return true;
    }
};

/**
 * Your Trie object will be instantiated and called as such:
 * Trie* obj = new Trie();
 * obj->insert(word);
 * bool param_2 = obj->search(word);
 * bool param_3 = obj->startsWith(prefix);
 */

File: 0208-implement-trie-prefix-tree/0208-implement-trie-prefix-tree.py
class Trienode:
    def __init__(self):
        self.children = [None]*26
        self.wordend = False    
        
class Trie:
        
    def __init__(self):
        self.root = Trienode()

    def insert(self, word: str) -> None:
        node = self.root
        
        for i in word:
            if not node.children[ord(i)-ord('a')]:
                node.children[ord(i)-ord('a')] = Trienode()
            node = node.children[ord(i)-ord('a')]
        
        node.wordend = True

    def search(self, word: str) -> bool:
        node = self.root
        
        for i in word:
            if not node.children[ord(i)-ord('a')]:
                return False
            node = node.children[ord(i)-ord('a')]
        
        return node.wordend

    def startsWith(self, prefix: str) -> bool:
        node = self.root
        
        for i in prefix:
            if not node.children[ord(i)-ord('a')]:
                return False
            node = node.children[ord(i)-ord('a')]
            
        return True


# Your Trie object will be instantiated and called as such:
# obj = Trie()
# obj.insert(word)
# param_2 = obj.search(word)
# param_3 = obj.startsWith(prefix)